src/lib/routers/history.ts

```diff
@@ -0,0 +1,197 @@
+import qs from 'qs';
+import { RouteState } from '../../types';
+
+type CreateUrlOptions = {
+  qsModule: typeof qs;
+  routeState: RouteState;
+  location: Location;
+};
+
+type ParseUrlOptions = {
+  qsModule: typeof qs;
+  location: Location;
+};
+
+type BrowserHistoryProps = {
+  windowTitle?: (routeState: RouteState) => string;
+  writeDelay: number;
+  createURL({ qsModule, routeState, location }: CreateUrlOptions): string;
```

Comment:
I was more thinking about extracting the whole function to be able to re-use it later in the file.

```ts
type CreateURL = ...
```